src/librustc_mir/interpret/terminator.rs

```diff
@@ -272,7 +261,7 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                 // No stack frame gets pushed, the main loop will just act as if the
                 // call completed.
                 if ret.is_some() {
```

Comment:
Make this if let some and use go_to_block

Reply:
This entire code here makes little sense actually... I'll adjust `call_intrinsic` to put responsibility for advancing the instruction pointer onto the hook, like we do for other `call_*` hooks.

Reply:
I decided to change this and, uh, made the PR a lot bigger along the way... but also made the APIs a lot better, I think.